sensor/common/compliance/nodescan_handler_test.go

```diff
@@ -89,47 +107,148 @@ func (s *NodeScanHandlerTestSuite) startForwardingToCentral(c <-chan *central.Ms
 // Exec with: go test -race -count=1 -v -run ^TestNodeScanHandler$ ./sensor/common/compliance
 func (s *NodeScanHandlerTestSuite) TestStopHandler() {
 	nodeScans := make(chan *storage.NodeScanV2)
+	defer close(nodeScans)
+	producer := newStoppable()
+	errTest := errors.New("example-stop-error")
 	h := NewNodeScanHandler(nodeScans)
-	// consume all messages send toCentral
+	s.NoError(h.Start())
+	consumer := consumeAndCount(h.ResponsesC(), 1)
+	// This is a producer that stops the handler after producing the first message and then sends many (29) more messages.
+	// This is to test whether the handler can shutdown with no leaks when the producer keeps producing messages.
 	go func() {
-		for {
+		defer producer.signalStopped()
+		for i := 0; i < 30; i++ {
 			select {
-			case <-s.ctx.Done():
+			case <-producer.stopC.Done():
 				return
-			case <-s.toCentral:
+			case nodeScans <- fakeNodeScanV2("Node"):
+				if i == 0 {
+					h.Stop(errTest)
+				}
```

Comment:
Just move waiting for counting consumer inside here and it will make sure the consumer sees its 1 message.
```suggestion
				if i == 0 {
					s.NoError(consumer.stoppedC.Wait()) // This blocks until consumer receives its 1 message
					h.Stop(errTest)
				}
```